lib/system/basic_types.nim

```diff
@@ -50,6 +50,13 @@ proc `and`*(x, y: bool): bool {.magic: "And", noSideEffect.}
   ## are true).
   ##
   ## Evaluation is lazy: if ``x`` is false, ``y`` will not even be evaluated.
+  ## Semantic check is lazy in VM, to allow `when declared(Foo) and T is Foo`
+
+template nimInternalAndVM*(x: bool, y: untyped): bool =
```

Comment:
This new way of "fixing" things by adding implementation details to system.nim is not acceptable.